src/Publisher.ts

```diff
@@ -39,8 +39,28 @@ export interface Asset {
 export interface Assets {
 	images: Array<Asset>;
 }
+
+export type TFrontmatter = Record<string, unknown> & {
+	"dg-path"?: string;
+	"dg-permalink"?: string;
+	"dg-home"?: boolean;
+	"dg-hide-in-graph"?: boolean;
+	"dg-hide"?: boolean;
+	"dg-pinned"?: boolean;
+	"dg-metatags"?: string;
+	tags?: string;
+};
+
+export type TPublishedFrontMatter = Record<string, unknown> & {
+	tags?: string[];
+	metatags?: string;
+	pinned?: boolean;
+	permalink?: string;
+	hide?: boolean;
+};
```

Comment:
let's improve on these, but also good to document here what they do for internal documentation. 

Later abstract the cache thing to these types to fix the final few type errors (and use these to simplify complex code)